nni/retiarii/oneshot/pytorch/base_lightning.py

```diff
@@ -0,0 +1,404 @@
+                            ' {"scheduler": scheduler, "monitor": "your_loss"}}'
+                        )
+                    is_one_cycle = isinstance(scheduler["scheduler"], optim.lr_scheduler.OneCycleLR)
+                    if is_one_cycle and scheduler.get("interval", "epoch") == "epoch":
+                        warn(
+                            "A `OneCycleLR` scheduler is using 'interval': 'epoch'."
+                            " Are you sure you didn't mean 'interval': 'step'?",
+                            RuntimeWarning,
+                        )
+                    lr_schedulers.append({**default_config, **scheduler})
+                elif isinstance(scheduler, optim.lr_scheduler.ReduceLROnPlateau):
+                    if monitor is None:
+                        raise Exception(
+                            "`configure_optimizers` must include a monitor when a `ReduceLROnPlateau`"
+                            " scheduler is used. For example:"
+                            ' {"optimizer": optimizer, "lr_scheduler": scheduler, "monitor": "metric_to_track"}'
+                        )
+                    lr_schedulers.append(
+                        {**default_config, "scheduler": scheduler, "reduce_on_plateau": True, "monitor": monitor}
+                    )
+                elif isinstance(scheduler, optim.lr_scheduler._LRScheduler):
+                    lr_schedulers.append({**default_config, "scheduler": scheduler})
+                else:
+                    raise ValueError(f'The provided lr scheduler "{scheduler}" is invalid')
+        return lr_schedulers
+
+    def configure_optimizers(self):
+        """
+        Combine architecture optimizers and user's model optimizers.
+        Overwrite configure_architecture_optimizers if architecture optimizers
+        are needed in your NAS algorithm.
+        By default ``self.model`` is currently a _SupervisedLearningModule in
+        nni.retiarii.evaluator.pytorch.lightning, and it only returns 1 optimizer.
+        But for extendibility, codes for other return value types are also implemented.
+        """
+        # pylint: disable=assignment-from-none
+        arc_optimizers = self.configure_architecture_optimizers()
+        if arc_optimizers is None:
+            return self.model.configure_optimizers()
+
+        if isinstance(arc_optimizers, optim.Optimizer):
+            arc_optimizers = [arc_optimizers]
+        self.arc_optim_count = len(arc_optimizers)
+
+        # The third return value ``frequency`` and ``monitor`` are ignored since lightning
+        # requires len(optimizers) == len(frequency), and gradient backword
+        # is handled manually.
+        w_optimizers, lr_schedulers, self.frequencies, monitor = \
+            self._configure_optimizers(self.model.configure_optimizers())
+        lr_schedulers = self._configure_schedulers(lr_schedulers, monitor, not self.automatic_optimization)
```

Comment:
Better to write comments/examples here to annotate what `w_optimizers`, `lr_schedulers`, `frequencies` and `monitor` look like.